x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -205,21 +210,27 @@ export const ComplianceDashboard = () => {
         preferredDashboard = KSPM_POLICY_TEMPLATE;
       }
       // cspm is installed
-      else if (installedPolicyTemplates?.includes(CSPM_POLICY_TEMPLATE)) {
+      else if (
+        installedPolicyTemplatesCspm !== 'unprivileged' &&
+        installedPolicyTemplatesCspm !== 'not-installed'
+      ) {
         preferredDashboard = CSPM_POLICY_TEMPLATE;
       }
       // kspm is installed
-      else if (installedPolicyTemplates?.includes(KSPM_POLICY_TEMPLATE)) {
+      else if (
+        installedPolicyTemplatesKspm !== 'unprivileged' &&
+        installedPolicyTemplatesKspm !== 'not-installed'
+      ) {
         preferredDashboard = KSPM_POLICY_TEMPLATE;
       }
-
       setSelectedTab(preferredDashboard);
     };
     selectInitialTab();
   }, [
     getCspmDashboardData.data?.stats.totalFindings,
     getKspmDashboardData.data?.stats.totalFindings,
-    getSetupStatus.data?.installedPolicyTemplates,
+    getSetupStatus.data?.cspm?.status,
+    getSetupStatus.data?.kspm?.status,
   ]);
```

Comment:
Improvement suggestion: move all that logic to its own function, `getInitialTab()`